### PR TITLE
connectec nest js with mongoose and initialised the server with compl…

### PR DESCRIPTION
**Pull request by Devesh21700Kumar**
…ete setup of User models modules controllers services

**Description**
onnected nest js with mongoose and initialised the 
server with complete setup of User models modules controllers services

Fixes # [ISSUE]

- [x] Code
- [x] New Feature

### CODE REVIEW

**Comment by Devesh21700Kumar** [comment]
@Abhishek-kumar09 i've made a complete user nestsjs setup and also added a mongoose connection

would dockerize the db container and setup the server soon after the merge

File: src/schemas/user.schema.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+export const UserSchema = new mongoose.Schema({
+  first_name: String,
+  last_name: String,
+  email: String,
+  phone: String,
+  address: String,
+  description: String,
+  created_at: { type: Date, default: Date.now },
+});
```
Comment (Abhishek-kumar09) [inline src/schemas/user.schema.ts:11]: Create schema with using `nestjs/mongoose`
https://docs.nestjs.com/techniques/mongodb

File: src/user/interfaces/user.interface.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { Document } from 'mongoose';
```
Comment (Abhishek-kumar09) [inline src/user/interfaces/user.interface.ts:1]: We will be using nestjs/mongoose and planning to remove `mongoose` completely.

**Review by Abhishek-kumar09 — CHANGES_REQUESTED** [review]
Some Items:

File: src/schemas/user.schema.ts (hunk 1 above)
Comment (Devesh21700Kumar) [reply src/schemas/user.schema.ts:11]: okay sure

File: src/user/interfaces/user.interface.ts (hunk 2 above)
Comment (Devesh21700Kumar) [reply src/user/interfaces/user.interface.ts:1]: sure

File: src/schemas/user.schema.ts (hunk 1 above)
Comment (Devesh21700Kumar) [reply src/schemas/user.schema.ts]: @Abhishek-kumar09 made this as per @nestjs/mongoose theme

File: src/user/interfaces/user.interface.ts (hunk 2 above)
Comment (Devesh21700Kumar) [reply src/user/interfaces/user.interface.ts:1]: @Abhishek-kumar09 we can't remove mongoose completely.. because @nestjs/mongoose is just a superset of it and for using document we would need mongoose

File: src/config/firebase.ts [hunk 3]
```diff
@@ -1,9 +0,0 @@
-import admin from 'firebase-admin';
-
-import * as firebaseAccountCredentials from './service.json';
-
-const serviceAccount = firebaseAccountCredentials as admin.ServiceAccount;
-
-export default {
-  credential: admin.credential.cert(serviceAccount),
-};
```
Comment (Abhishek-kumar09) [inline src/config/firebase.ts:9]: Why is this changed?

Comment (Devesh21700Kumar) [reply src/config/firebase.ts:9]: service.json doesnot exist and this file was not imported anywhere else

Comment (Devesh21700Kumar) [reply src/config/firebase.ts:9]: so this was giving error

Comment (Devesh21700Kumar) [reply src/config/firebase.ts]: @Abhishek-kumar09 it seems like i had forgot to set up the key.. now i've restored this by adding the service.json

File: .eslintignore [hunk 4]
```diff
@@ -0,0 +1 @@
+/src/user/user.controller.ts
```
Comment (ritvij14) [inline .eslintignore:1]: I think we should revert this. This will prevent user.controller.ts from being checked by the linter. We don't want that.

File: .eslintrc.js [hunk 5]
```diff
@@ -20,6 +20,7 @@ module.exports = {
     '@typescript-eslint/explicit-function-return-type': 'off',
     '@typescript-eslint/explicit-module-boundary-types': 'off',
     '@typescript-eslint/no-explicit-any': 'off',
+    "@typescript-eslint/no-unused-vars": 'off',
```
Comment (ritvij14) [inline .eslintrc.js:23]: Right below this statement is the one that turns this from a warning to an error. Either delete that or remove yours. I was asked to add that in #20

**Review by ritvij14 — CHANGES_REQUESTED** [review]

File: .eslintrc.js (hunk 5 above)
Comment (Abhishek-kumar09) [reply .eslintrc.js:23]: remove this line, instead just do that is causing the error: https://github.com/codeforcauseorg/edu-server/commit/9d83abab7025c8e4496e532a8271174918b59825

File: package-lock.json [hunk 6]
```diff
@@ -1,8 +1,10906 @@
...
   "version": "0.0.1",
-  "lockfileVersion": 1,
+  "lockfileVersion": 2,
   "requires": true,
+  "packages": {
+    "": {
+      "name": "edu-client",
+      "version": "0.0.1",
+      "license": "MIT",
+      "dependencies": {
```
Comment (Abhishek-kumar09) [inline package-lock.json:11]: Please dont add unwanted lock file changes. Just those changes are needed where package is installed.

What you should do is just copy the lock file as it is from the main repo and do npm install again. It will just change the required lines in lock file.

File: .eslintrc.js (hunk 5 above)
Comment (Devesh21700Kumar) [reply .eslintrc.js:23]: image


Even after using UserService and CreateUserDTO no-used-vars error appeared.. that's why i wanted to remove that line @Abhishek-kumar09

File: package-lock.json (hunk 6 above)
Comment (Devesh21700Kumar) [reply package-lock.json]: @Abhishek-kumar09 restored the original package-lock.json

**Comment by Abhishek-kumar09** [comment]
This is not required: 
image

File: README.md [hunk 7]
```diff
@@ -12,7 +12,7 @@
 
 ## Description
 
-Education platform is a product which will enable 100% education and will help in upskilling developing countries and rural areas.
+Education platform is a product which will strive to enable 100% education and will help in upskilling developing countries and rural areas.
 Edu server is a backend application written using Nestjs, it provides api endpoints which are used by [Edu Client](https://github.com/codeforcauseorg/edu-client) mobile application.
 
 ### Table of Contents
```
Comment (Abhishek-kumar09) [inline README.md:18]: Add to readme how to setup mongodb also in brief.

**Comment by Abhishek-kumar09** [comment]
@Devesh21700Kumar Please add documentation for this one how one can setup mongodb with the app
